ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -185,24 +342,116 @@ public void ExportToFile(string resourceName, string fileName, VMesh mesh)
         {
             DstDir = Path.GetDirectoryName(fileName);
 
-            var exportedModel = ModelRoot.CreateModel();
-            exportedModel.Asset.Generator = GENERATOR;
+            var exportedModel = CreateModelRoot(resourceName, out var scene);
             var name = Path.GetFileName(resourceName);
-            var scene = exportedModel.UseScene(name);
+            var node = AddMeshNode(exportedModel, scene, null, name, mesh, null);
+
+            if (node != null)
+            {
+                // Swap Rotate upright, scale inches to meters.
+                node.WorldMatrix = TRANSFORMSOURCETOGLTF;
+            }
+
+            WriteModelFile(exportedModel, fileName);
+        }
 
-            var exportedMesh = CreateGltfMesh(name, mesh, exportedModel, false);
-            var meshNode = scene.CreateNode(name)
-                .WithMesh(exportedMesh);
+        private Node AddMeshNode(ModelRoot exportedModel, Scene scene, VModel model, string name,
+            VMesh mesh, Skeleton skeleton, string skinMaterialPath = null)
+        {
+            if (mesh.GetData().GetArray("m_sceneObjects").Length == 0)
+            {
+                return null;
+            }
+
+            var hasJoints = skeleton != null && skeleton.AnimationTextureSize > 0;
+
+            if (MeshNameCounterDict.TryGetValue(name, out var counter))
```

Comment:
What problem is this solving exactly?

Reply:
If multiple instances of the same mesh exist, instead of exporting a copy of the mesh in the glTF each time, export it once and then add references to it for the duplicates.

Reply:
Proposal to improve this: Get rid of the counter and just store a name -> node mapping in the dictionary. Then this if statement just becomes:
```cs
if (LoadedMeshDictionary.TryGetValue(name, out var existingNode)) {
    if (!hasJoints && existingNode.Skin == null)
    {
        var newNode = scene.CreateNode(name);
        newNode.Mesh = existingNode.Mesh;
        return newNode;
    }
}
 ```

Reply:
Usage of the counter was to keep the node names unique, but the glTF spec doesn't seem to care, so I went ahead with this change. Thanks!